hooks/helpers/useHandleError.ts

```diff
@@ -1,16 +1,48 @@
+import { errorFetchingVoteDataMessage, wrongChainMessage } from "constant";
+import { formatTransactionError } from "helpers";
 import { useErrorContext } from "hooks";
 import { ErrorOriginT } from "types";
 
-export function useHandleError(errorOrigin?: ErrorOriginT) {
-  const { addErrorMessage } = useErrorContext(errorOrigin);
+interface Options {
+  errorOrigin?: ErrorOriginT;
+  isContractTransaction?: boolean;
+  isDataFetching?: boolean;
+  customErrorMessage?: string;
+}
+export function useHandleError(options?: Options) {
+  const {
+    errorOrigin,
+    isContractTransaction = true,
+    isDataFetching = false,
+    customErrorMessage,
+  } = options || {};
+  const { addErrorMessage, clearErrorMessages, errorMessages } =
+    useErrorContext(errorOrigin);
+
+  function onError(error: unknown) {
+    if (errorMessages.includes(wrongChainMessage)) return;
+
+    let message: string;
 
-  return (error: unknown) => {
-    if (error instanceof Error) {
-      addErrorMessage(error.message);
+    if (isDataFetching) {
+      message = errorFetchingVoteDataMessage;
```

Comment:
Instead of showing a different error for every data fetching error that happens, show one generic error message. Since data fetching either works or it doesn't (we need to load all the vote data from the chain for the dapp to work) there's no point in showing 10 different errors that all say some variation of "error fetching x".